### PR TITLE
Improve logic to retrieve sort name

### PR DESCRIPTION
**Pull request by reginafcompton**
Handles issue #269

### CODE REVIEW

File: lametro/utils.py [hunk 1]
```diff
@@ -16,24 +17,17 @@ def format_full_text(full_text):
...
-# Isolate text after 'SUBJECT'
 def parse_subject(text):
-    if text:
-        before_keyword, keyword, after_keyword = text.partition('SUBJECT:')
-        if after_keyword:
-            if '[PROJECT OR SERVICE NAME]' not in after_keyword and '[DESCRIPTION]' not in after_keyword and '[CONTRACT NUMBER]' not in after_keyword:
-                return after_keyword.strip()
-
-    return None
+    if ('[PROJECT OR SERVICE NAME]' not in text) and ('[DESCRIPTION]' not in text) and ('[CONTRACT NUMBER]' not in text):
```
Comment (hancush) [inline lametro/utils.py]: Great improvement. :-)

File: lametro/utils.py [hunk 2]
```diff
@@ -16,24 +17,17 @@ def format_full_text(full_text):
...
-        txt_as_array = full_text.split("..")
-        for item in txt_as_array:
-            if 'SUBJECT:' in item:
-                array_with_subject = item.split('\n\n')
-                for item in array_with_subject:
-                    if 'SUBJECT:' in item:
-                        results = item.replace('\n', '')
+        full_text_with_cue = full_text.replace('\n\n', 'NEWLINE').replace('\r\n', 'NEWLINE').replace('\n..', 'NEWLINE')
+
+        match = re.search('(SUBJECT:)(.*?)(NEWLINE|ACTION:)', full_text_with_cue.replace('\n', ' '))
```
Comment (hancush) [inline lametro/utils.py]: Why do this replace here, instead of in the `full_text_with_cue` definition?

**Comment by hancush** [comment]
@reginafcompton I see your point, but if we are debugging an unanticipated parsing failure, I wouldn't immediately think to check the tests for examples. I also think it's kind of hard to follow regexes, if I don't know what the string might look like.

File: lametro/utils.py (hunk 2 above)
Comment (reginafcompton) [reply lametro/utils.py]: Yes, it seems like something like this would be better:

```
clean_full_text = full_text.replace('\n\n', 'NEWLINE').replace('\r\n', 'NEWLINE').replace('\n..', 'NEWLINE').replace('\n', ' ')

match = re.search('(SUBJECT:)(.*?)(NEWLINE|ACTION:)', clean_full_text)
```

Honestly, I think there must be a cleverer use of RegEx to get the desired results. Have you had any luck with MULTILINE? https://docs.python.org/3.6/library/re.html#re.MULTILINE That presumably would eliminate the need for `replace`....but it did not behave as I expected.

Comment (hancush) [reply lametro/utils.py]: you know, i'm really terrible at regex and [tend to avoid them](http://regex.info/blog/2006-09-15/247).

what is the intent here? it seems like you are just looking for "SUBJECT" and either "NEWLINE" or "ACTION".

could you do `'SUBJECT' in clean_full_text and any(token in clean_full_text for token in ['NEWLINE', 'ACTION'])` or something like that?

Comment (hancush) [reply lametro/utils.py]: are new lines meaningful?

Comment (reginafcompton) [reply lametro/utils.py]: Yes, I fear RegEx, too. [My previous solution](https://github.com/datamade/la-metro-councilmatic/blob/master/lametro/utils.py#L15) completely avoided RegEx, but it also missed some edge cases and was more difficult to decipher. 

I added some comments to give you a sense about how newlines help parse the subject header. @hancush - happy to chat about it!

File: lametro/utils.py [hunk 3]
```diff
@@ -11,29 +12,33 @@
...
     if full_text:
-        txt_as_array = full_text.split("..")
-        for item in txt_as_array:
-            if 'SUBJECT:' in item:
-                array_with_subject = item.split('\n\n')
-                for item in array_with_subject:
-                    if 'SUBJECT:' in item:
-                        results = item.replace('\n', '')
+        clean_full_text = full_text.replace('\n\n', 'NEWLINE').replace('\r\n', 'NEWLINE').replace('\n..', 'NEWLINE').replace('\n', ' ')
+        match = re.search('(SUBJECT:)(.*?)(NEWLINE|ACTION:)', clean_full_text)
```
Comment (hancush) [inline lametro/utils.py]: I think moving that last replace to the line above (and seeing examples in the docstring!) really helped with the clarity here. Text parsing sucks, haha. Nice changes, all around.
